erdos/src/bin/erdos.rs

```diff
@@ -0,0 +1,163 @@
+use tokio::{
+    sync::mpsc::{self, Receiver},
+    task::JoinHandle,
+};
+use tracing::Level;
+
+#[derive(Debug)]
+enum CLINotifications {
+    Shutdown,
+}
+
+/// The return type to be used when setting up notification channels between
+/// the signal handlers and the main Node loop.
+type NotificationChannel = (Receiver<CLINotifications>, Handle, JoinHandle<()>);
+
+fn setup_notification_channels() -> Result<NotificationChannel, Box<dyn std::error::Error>> {
+    // Initialize the Signals handled by the CLI.
+    let mut signals = Signals::new(&[SIGINT])?;
+    let signals_handle = signals.handle();
+
+    // Initialize a Broadcast channel to be used to talk to the nodes.
+    let (driver_notification_tx, driver_notification_rx) = mpsc::channel(100);
+    let signals_task = tokio::spawn(async move {
+        while let Some(signal) = signals.next().await {
+            match signal {
+                SIGINT => {
+                    driver_notification_tx
+                        .send(CLINotifications::Shutdown)
+                        .await
+                        .unwrap();
+                }
+                _ => unreachable!(),
+            }
+        }
+    });
+    Ok((driver_notification_rx, signals_handle, signals_task))
+}
+
+async fn cleanup_notification_channels(
+    signals_handle: Handle,
+    signals_task: JoinHandle<()>,
+) -> Result<(), Box<dyn std::error::Error>> {
+    // Uninstall the signal handlers, and wait for the signal task to complete.
+    signals_handle.close();
+    signals_task.await?;
+    Ok(())
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let matches = App::new(env!("CARGO_PKG_NAME"))
+        .version(env!("CARGO_PKG_VERSION"))
+        .author(env!("CARGO_PKG_AUTHORS"))
+        .about(env!("CARGO_PKG_DESCRIPTION"))
+        .subcommand(
+            SubCommand::with_name("start")
+                .about("Initialize a Leader or a Worker node.")
+                .arg(
+                    Arg::with_name("leader")
+                        .long("leader")
+                        .takes_value(false)
+                        .help("Initialize a Leader node."),
+                )
+                .arg(
+                    Arg::with_name("address")
+                        .long("address")
+                        .default_value("0.0.0.0:4444")
+                        .help("The address that is used by the Leader node."),
+                )
+                .arg(
+                    Arg::with_name("num_gpus")
+                        .long("num_gpus")
+                        .default_value("0")
+                        .help("The number of GPUs on this Worker node."),
+                )
+                .arg(
+                    Arg::with_name("num_cpus")
+                        .long("num_cpus")
+                        .default_value("0")
+                        .help("The number of CPUs on this Worker node."),
+                )
```

Comment:
This was in spirit of [Ray](https://docs.ray.io/en/latest/ray-core/configure.html#configuring-ray) which provides separate flags for CPUs and GPUs, and then a general JSON string to be parsed for custom resources.

The Resources abstraction is a WIP and the idea was to leave it at the basic level right now. 

The `bin/erdos.rs` needs to be reworked anyways since `Worker`s cannot be spawned via the CLI pending a solution to distributing the `JobGraph`.